DB設計

ooba-noriyuki opened the pull request:
# What
README.mdにChatSpaceに必要なデータベース設計を記述する。

# Why
ChatSpaceの開発に必要なデータベースのテーブルとアソシエーションの関係を明確にするため。

ooba-noriyuki committed:
Edit README.md

DB設計

ooba-noriyuki committed:
Edit README.md

マークダウン記述の修正

スペース追加

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,56 @@
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false|
+|email|string|null: false,uniqe: true|
```
カンマの後に半角スペース入れましょう。

ooba-noriyuki replied:
修正しました。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,56 @@
+|image|string|
+|group_id|references|null: false, foreign_key: true|
+|user_id|references|null: false, foregin_key: true|
+|timestamps|datetime|null: false|
```
timestampsはデフォルトでどのテーブルにも付くので今回記述しなくて大丈夫です！

ooba-noriyuki replied:
削除しました。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,56 @@
+
+### Association
+- has_many :users, through: :members
+- has_many :members
```
has_many :users, through: :membersよりも先に記述しましょう。
そうしないとアソシエーションで上手く行かないです。

ooba-noriyuki replied:
修正しました。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,56 @@
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false|
+|user_id|references|null: false, foreign_key: true|
```
こちらは中間テーブルを介して参照しますので不要です。

ooba-noriyuki replied:
削除しました。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,56 @@
+### Table Definition
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false|
```
後半にnameを検索する機能を付けるのでindexオプションを付けましょう。

MrUrakou replied:
こちらいかがですか？

ooba-noriyuki replied:
修正しました。

ooba-noriyuki committed:
Edit README.md

誤記訂正（usersテーブル）
不要なカラムの削除（messagesテーブル、groupsテーブル）
アソシエーションの記述の順番の修正（groupsテーブル）
indexオプションの追加（usersテーブル）

quelmecabj commented:
LGTMです！
